src/environments.js

```diff
@@ -66,10 +66,12 @@ var environmentDefinitions = [
             var parser = this;
             // Currently only supports alignment, no separators like | yet.
             colalign = colalign.value.map ? colalign.value : [colalign];
-            colalign = colalign.map(function(node) {
+            var cols = colalign.map(function(node) {
                 var ca = node.value;
                 if ("lcr".indexOf(ca) !== -1) {
-                    return ca;
+                    return {
+                        align: ca
+                    };
```

Comment:
It seems like ```@{...}``` can come before or after a column specifier, the same goes for ```|```.  Maybe instead of grouping these things with each column, ```cols``` could become ```colSpec``` which could contain separate entries for alignment, spacing, and vertical bar specifiers, e.g. ```{|r|@{\quad}c@{\quad}}``` would be represented by:
```
var colSpec = [
   { type: 'sep', value: '|' },
   { type: 'align', value: 'r' },
   { type: 'space', value: quad },
   { type: 'align', value: 'c' },
   { type: 'space', value: quad }
];
```
Thoughts?